float64 d1
float64 d2
float64 d3
float64 d4
float64 d5
float64 d6
